presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkStaticQueryExecution.java

```diff
@@ -159,6 +160,14 @@ protected List<Tuple2<MutablePartitionId, PrestoSparkSerializedPage>> doExecute(
             throws SparkException, TimeoutException
     {
         SubPlan rootFragmentedPlan = createFragmentedPlan();
+
+        // executor allocation is currently only supported at root level of the plan
+        // in future this could be extended to fragment level configuration
+        if (planAndMore.getPhysicalResourceSettings().getMaxExecutorCount().isPresent()) {
+            sparkContext.sc().conf().set(SPARK_DYNAMIC_ALLOCATION_MAX_EXECUTORS_CONFIG,
+                    Integer.toString(planAndMore.getPhysicalResourceSettings().getMaxExecutorCount().orElse(0)));
```

Comment:
I guess having `0` value for maxExecutor should not be allowed. We should add a validation check to ensure that its never `0` and remove the `orElse` part here.

Reply:
Actually, `.orElse()` is not even requires, as this is already handled maxExecutorCount being `Optional<Intger>`.
This code block won't be triggered if the value is not present (hence use default value).